api/data/refactoring.py

```diff
@@ -154,3 +158,118 @@ def save_lovd_as_vcf(data, save_to="./lovd.vcf"):
 
             f.write("\t".join(record))
             f.write("\n")
+
+
+def get_variant_ids_from_clinvar_name_api(name: str, count: int = 100):
+    """
+    Extracts variant ids from ClinVar `name` variable. /n
+    key of dictionary is the size of the list of ids.
+
+    :param str name: name of variant
+    :param int count: number of ids to extract
+    :returns: ids of variants
+    :rtype: str
+    """
+
+    result = {}
+
+    separator = ","
+    clinvar_url = f"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=clinvar&term={name}&retmode=json&retmax={count}"
+
+    request = requests.get(clinvar_url)
+
+    if request.status_code != 200:
+        raise ValueError(f"Request failed with status code {request.status_code}")
+
+    data = request.json()
+
+    ids = data['esearchresult']['idlist']
+
+    result['idlist'] = ids
+    result['count'] = data['esearchresult']['count']
+
+    return result
+
+
+def request_clinvar_api_data(gene_id: str):
+    """
+    Requests ClinVar API for data about variant with given id.
+    Converts it to pandas dataframe.
+
+    :param str gene_id: id of variant (may be multiple)
+    :returns: dataframe from ClinVar API
+    :rtype: dataframe
+    """
+    clinvar_url = f"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=clinvar&id={gene_id}&retmode=json"
+
+    request = requests.get(clinvar_url)
+
+    if request.status_code != 200:
+        raise ValueError(f"Request failed with status code {request.status_code}")
+
+    data = request.json()
+
+    results = data['result']
```

Comment:
make as single line, data is used only once. Better to assigne to `data`, because `results` is not good naming for variable if it's not an output. Anything can be `result` in the process